plugins/backstage-plugin-coder/docs/guides/coder-api.md

```diff
@@ -0,0 +1,257 @@
+# Coder API - Quick-start guide
+
+## Overview
+
+The Coder plugin makes it easy to bring the entire Coder API into your Backstage deployment. This guide covers how to get it set up so that you can start accessing Coder from Backstage.
+
+Note: this covers the main expected use cases with the plugin. For more information and options on customizing your Backstage deployment further, see our [Advanced API guide](./coder-api-advanced.md).
+
+### Before you begin
+
+Please ensure that you have the Coder plugin fully installed before proceeding. You can find instructions for getting up and running in [our main README](../../README.md).
+
+### Important hooks for using the Coder API
+
+There are a few React hooks that are needed to interact with the Coder API. These can be split into three categories: React Query hooks, core plugin hooks, and convenience hooks.
+
+#### React Query hooks
+
+The Coder plugin uses [React Query/TanStack Query v4](https://tanstack.com/query/v4/docs/framework/react/overview) for all of its data caching. We recommend that you use it for your own data caching, because of the sheer amount of headaches it can spare you.
+
+There are three main hooks that you will likely need:
+
+- `useQuery` - Query and cache data
+- `useMutation` - Perform mutations on an API resource
+- `useQueryClient` - Coordinate queries and mutations
+
+#### Core plugin hooks
+
+These are hooks that provide direct access to various parts of the Coder API.
+
+- `useCoderApi` - Exposes an object with all available Coder API methods. For the most part, there is no exposed state on this object; you can consider it a "function bucket".
+- `useCoderAuth` - Provides methods and state values for interacting with your current Coder auth session from within Backstage.
+
+#### Convenience hooks
+
+- `useCoderQuery` - Simplifies wiring up `useQuery`, `useCoderApi`, and `useCoderAuth`
+
+## Recommendations for accessing the API
+
+1. If querying data, prefer `useCoderQuery`. It automatically wires up all auth logic to React Query (which includes pausing queries if the user is not authenticated). It also lets you access the Coder API via its query function. `useQuery` is also a good escape hatch if `useCoderQuery` doesn't meet your needs, but it requires more work to wire up correctly.
+2. If mutating data, you will need to call `useMutation`, `useQueryClient`, and `useCoderApi` in tandem\*. The plugin exposes a `CODER_QUERY_KEY_PREFIX` constant that you can use to group all Coder queries together.
+
+We highly recommend **not** fetching with `useState` + `useEffect`, or with `useAsync`. Both face performance issues when trying to share state. See [ui.dev](https://www.ui.dev/)'s wonderful [_The Story of React Query_ video](https://www.youtube.com/watch?v=OrliU0e09io) for more info on some of the problems they face.
```

Comment:
Any reason we have to include this disclaimer? Seems like general React knowledge, rather than Coder plugin specific knowledge.

Reply:
That makes sense! Could we just move it after the meat of the examples then?

Reply:
Honestly, I would expect a good chunk of plugin users not to be super seasoned with React